Documentation/VisualThemes.md

```diff
@@ -0,0 +1,281 @@
+
+[Theme configurations](#theme-configuration) are definitions of properties and types while [Theme Engines](#theme-engines) are classes that consume the configurations and implement the logic to update transforms, materials, and more at runtime.
+
+## Theme configuration
+
+Theme configurations are [ScriptableObjects](https://docs.unity3d.com/Manual/class-ScriptableObject.html) that define how Theme Engines will be initialized at runtime. They define what properties and values to utilize in response to input or other state changes when the app is running. As [ScriptableObjects](https://docs.unity3d.com/Manual/class-ScriptableObject.html) assets, theme configurations can be defined once and then re-used across different UX components.
+
+To create a new [`Theme`](xref:Microsoft.MixedReality.Toolkit.UI.Theme) asset:
+
+1) Right click in the *Project Window*
+1) Select **Create** > **Mixed Reality Toolkit** > **Theme**
+
+Example Theme configuration assets can be found under [MixedRealityToolkit.SDK/Features/UX/Interactable/Themes](https://github.com/microsoft/MixedRealityToolkit-Unity/tree/mrtk_release/Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Themes).
+
+![Theme ScriptableObject example in inspector](Images/VisualThemes/ThemeInspectorExample.png)
+
+### States
+
+When creating a new [`Theme`](xref:Microsoft.MixedReality.Toolkit.UI.Theme), the first thing to set is what states are available. The *States* property indicates how many values a Theme configuration needs to define as there will be one value per state. In the example image above, the default states defined for the [Interactable](README_Interactable.md) component are *Default*, *Focus*, *Pressed*, and *Disabled*. These are defined in the [DefaultInteractableStates](https://github.com/microsoft/MixedRealityToolkit-Unity/tree/mrtk_release/Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/States/DefaultInteractableStates.asset) asset file.
+
+To create a new [`State`](xref:Microsoft.MixedReality.Toolkit.UI.States) asset:
+
+1) Right click in the *Project Window*
+1) Select **Create** > **Mixed Reality Toolkit** > **State**
+
+![States ScriptableObject example in inspector](Images/Interactable/DefaultInteractableStates.png)
+
+A [`State`](xref:Microsoft.MixedReality.Toolkit.UI.States) ScriptableObject defines both the list of states as well as the type of *StateModel* to create for these states. A *StateModel* is a class that extends [`BaseStateModel`](xref:Microsoft.MixedReality.Toolkit.UI.BaseStateModel) and implements the state machine logic to generate the current state at runtime. The current state from this class is generally used by Theme Engines at runtime to dictate what values to set against material properties, GameObject transforms, and more.
```

Comment:
nit: can you also add a link to "State" documentation here  - pointing to the blurb in README_Interactable.md please.